sparse/src/KokkosSparse_Utils.hpp

```diff
@@ -848,6 +848,21 @@ ordinal_t graph_max_degree(const rowmap_t &rowmap) {
   return val;
 }
 
+template <typename execution_space, typename rowmap_t>
+typename rowmap_t::non_const_value_type graph_max_degree(
+    const execution_space &exec, const rowmap_t &rowmap) {
+  using Offset  = typename rowmap_t::non_const_value_type;
+  using Reducer = Kokkos::Max<Offset>;
+  Offset nrows  = rowmap.extent(0);
+  if (nrows) nrows--;
```

Comment:
What if an empty matrix is passed?
Then you will get `nrows == -1` and that might trip the `RangePolicy`

Reply:
If rowmap has extent 0, then the branch ``if(nrows)`` won't get taken and nrows will stay at 0

Reply:
Good point, I miss read the logic here.